### ISSUE TITLE
[SR-4091] Incorrect result shown in REPL for 'UInt8?'

### ISSUE THREAD

**Issue by Bouke** [swiftlang/llvm-project#4394]
|                  |                 |
   |------------------|-----------------|
   |Previous ID       | SR-4091      |
   |Radar             | None         |
   |Original Reporter | @Bouke      |
   |Type              | Bug    |
   |Status            | Resolved        |
   |Resolution        | Done    |
        
<details>
  <summary>Environment</summary>

Feb 24, Swift 3.1 dev snapshot

</details>

   <details>
  <summary>Additional Detail from JIRA</summary>

   |                  |                 |
   |------------------|-----------------|
   |Votes             | 2         |
   |Component/s       | LLDB for Swift    |
   |Labels            | Bug, REPL        |
   |Assignee          | @dcci      |
   |Priority          | Medium      |

   

   md5: 019696cf1364b986ef151bb5fd4b0156

  </details>

**is duplicated by**:  
* [SR-4326](https://bugs.swift.org/browse/SR-4326) lldb prints strange numbers when interpreting optional unsigned values  
* [SR-9703](https://bugs.swift.org/browse/SR-9703) REPL has incorrect summaries of optional unsigned integers  

**relates to**:  
* [SR-11593](https://bugs.swift.org/browse/SR-11593) REPL incorrectly reports URL as nil  

**Issue Description:**

``` java
Welcome to Apple Swift version 3.1-dev (LLVM a7c680da51, Clang 2b4323722b, Swift 7e1648cd24). Type :help for assistance.
  1> UInt8("ff", radix: 16)
$R0: UInt8? = -1
  2> $R0!
$R1: UInt8 = 255
  3> UInt8("ff", radix: 16)!
$R2: UInt8 = 255
  4>
```

**Comment by CodaFi** [comment]
This one is an LLDB formatter problem.

**Comment by belkadan** [comment]
We were definitely supposed to have fixed it, though. SR-1007

**Comment by belkadan** [comment]
Ah, the Optional is significant.

**Comment by 0acf245c-f0fa-406f-afb9-4746d10f0359** [comment]
I'm going to take a look at this one tomorrow. Thanks

**Comment by 0acf245c-f0fa-406f-afb9-4746d10f0359** [comment]
Tomorrow ended up being 4 days later.

**Comment by 0acf245c-f0fa-406f-afb9-4746d10f0359** [comment]
This is what I see in the latest snapshot.

``` java
Welcome to Swift version 5.1.1-dev (LLVM 6e04008c7f, Swift 7032e652ee).
Type :help for assistance.
  1>  UInt8("ff", radix: 16)
$R0: UInt8? = 255
  2> $R0!
$R1: UInt8 = 255
  3> UInt8("ff", radix: 16)!
$R2: UInt8 = 255
```

**Comment by CodaFi** [comment]
@dcci This seems resolved. Do you know what would have changed to fix this?

**Comment by 0acf245c-f0fa-406f-afb9-4746d10f0359** [comment]
This is also \<rdar://problem/37835998\> and it was fixed by <https://github.com/apple/swift-lldb/pull/1793>

**Comment by 0acf245c-f0fa-406f-afb9-4746d10f0359** [comment]
@CodaFi I guess I just replied to you, Robert 🙂
